Switch YuvToRgbConverter to take in ImageProxy

rogerthat94 opened the pull request:
Change YuvToRgbConverter to take in an ImageProxy.

This allows us to remove the `@SuppressLint("UnsafeExperimentalUsageError")` annotation from `bindCameraUseCases`.

ImageProxy is returned by CameraX. Its [documentation](https://developer.android.com/reference/androidx/camera/core/ImageProxy) notes that it "has a similar interface as Image."

owahltinez commented:
With this change, you are adding CameraX as a dependency for the utils library (without adding it to the gradle file, so the standalone library project won't build).

This also prevents developers from using the converter if they are using Camera2 instead of CameraX. Instead of this, you could simply call `ImageProxy.getImage()` ([docs](https://developer.android.com/reference/androidx/camera/core/ImageProxy#getImage())).

rogerthat94 commented:
Calling `ImageProxy.getImage()` requires adding `@SuppressLint("UnsafeExperimentalUsageError")` to the function you do this in. See [here](https://github.com/android/camera-samples/blob/4aac9c7763c285d387194a558416a4458f29e275/CameraXTfLite/app/src/main/java/com/example/android/camerax/tflite/CameraActivity.kt#L140). 

`CameraXTfLite` is the only sample in this repository that uses this converter, and it already has CameraX as a dependency. Do you think it's worth having a version of the converter that doesn't use `ImageProxy.getImage()` so that `CameraXTfLite` (and other people who use the converter with CameraX) don't have to add `@SuppressLint("UnsafeExperimentalUsageError")`?

owahltinez commented:
> Calling ImageProxy.getImage() requires adding @SuppressLint("UnsafeExperimentalUsageError") to the function you do this in. See here.

That's just a lint annotation because CameraX marks the `ImageProxy.getImage()` method as experimental. Is the `@SuppressLint` declaration causing any issues for you?

> CameraXTfLite is the only sample in this repository that uses this converter, and it already has CameraX as a dependency

The library should also include the appropriate dependencies regardless of what projects use it.

rogerthat94 commented:
The `@SuppressLint` isn't causing issues. But it seems unnecessary when passing an ImageProxy would be an option. Doesn't CameraX mark `ImageProxy.getImage()` as experimental to discourage people from using it? 

Would you be opposed to adding a CameraX dependency to the utils library? If so, what are your thoughts on making a separate version of the converter somewhere else that uses ImageProxy instead of Image?

owahltinez commented:
Sorry for the late reply... I thought I had already replied to this, I guess I only did that in my head :-)

> But it seems unnecessary when passing an ImageProxy would be an option. Doesn't CameraX mark ImageProxy.getImage() as experimental to discourage people from using it?

It's not intended as discouragement, it only indicates that the behavior or the API might change in a breaking way in the future. Currently, what we show is the recommended to go from `ImageProxy` to `Image`.

> Would you be opposed to adding a CameraX dependency to the utils library? If so, what are your thoughts on making a separate version of the converter somewhere else that uses ImageProxy instead of Image?

At the moment, yes; because the utils library is being used by almost all samples and only 2 of them use CameraX. If you would like, I wouldn't be opposed to adding documentation saying "this could be changed to use ImageProxy instead" and even add a link to your commit!

rogerthat94 commented:
No problem!

> It's not intended as discouragement, it only indicates that the behavior or the API might change in a breaking way in the future.
Got it! 

I was misunderstanding what the "Unsafe" in "UnsafeExperimentalUsageError" meant. 

> I wouldn't be opposed to adding documentation saying "this could be changed to use ImageProxy instead" and even add a link to your commit!

I have added comments saying this. Android studio highlights `[Media.Image]` as an error in the document string, so I changed this to [Image].

ggfan commented:
@rogerthat94 could you access https://github.com/android/camera-samples/discussions/399?

ggfan commented:
probably not a good idea to handle this PR this way after chatted internally. some more discussion in #399, thank you for the PR!
